### PR TITLE
Fix corner cases for FileSink.flush

### PR DESCRIPTION
**Pull request by windie**
There are two corner cases for `flush`:

- No record has been processed before the first `flush`: it will output "]" to the stdout and close it.
- No record has been processed between two `flush`s: it will write "]" to a closed writer. Currently it's silent as PrintStream always swallows IOException.

This PR fixed `flush` by creating a new file for these corner cases.

### CODE REVIEW

File: heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/sink/FileSink.java [hunk 1]
```diff
@@ -64,25 +64,20 @@
 
   @Override
   public void init(Map<String, Object> conf, SinkContext context) {
-    filenamePrefix = (String) conf.get(FILENAME_KEY) + "." + context.getMetricsMgrId();
+    filenamePrefix = conf.get(FILENAME_KEY) + "." + context.getMetricsMgrId();
```
Comment (maosongfu) [inline heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/sink/FileSink.java]: Can we check and throw exceptions here if config does not contain "filename-output" or "file-maximum"? Or provide some defaults values?

Comment (windie) [inline heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/sink/FileSink.java]: What's the contract for conf? I saw most of places don't check the existence. E.g.,  https://github.com/twitter/heron/blob/master/heron/common/src/java/com/twitter/heron/common/config/SystemConfig.java#L468

Comment (maosongfu) [inline heron/metricsmgr/src/java/com/twitter/heron/metricsmgr/sink/FileSink.java]: Yes those places should have been checked.

**Comment by maosongfu** [comment]
Can a also add a unit test to verify the corner cases trying to solve in https://github.com/twitter/heron/tree/master/heron/metricsmgr/tests/java/com/twitter/heron/metricsmgr ?

**Comment by windie** [comment]
Updated. PTAL.

**Comment by maosongfu** [comment]
Thanks for the contribution. Looks good to me. :+1:
